### PR TITLE
[Validator] Use LogicException for missing Property Access Component in comparison constraints

### PR DESCRIPTION
**Pull request by Lctrs**
| Q             | A
| ------------- | ---
| Branch?       | 4.2
| Bug fix?      | yes-ish
| New feature?  | no
| BC breaks?    | yes-ish
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | N/A
| License       | MIT
| Doc PR        | N/A

For consistency.

### CODE REVIEW

**Review by fabpot — APPROVED** [review]

**Review by OskarStark — APPROVED** [review]

File: src/Symfony/Component/Validator/Constraints/AbstractComparison.php [hunk 1]
```diff
@@ -46,7 +47,7 @@ public function __construct($options = null)
             }
 
             if (isset($options['propertyPath']) && !class_exists(PropertyAccess::class)) {
-                throw new ConstraintDefinitionException(sprintf('The "%s" constraint requires the Symfony PropertyAccess component to use the "propertyPath" option.', \get_class($this)));
+                throw new LogicException(sprintf('The "%s" constraint requires the Symfony PropertyAccess component to use the "propertyPath" option.', \get_class($this)));
```
Comment (OskarStark) [inline src/Symfony/Component/Validator/Constraints/AbstractComparison.php]: No tests to adjust? 🧐

Comment (Lctrs) [reply src/Symfony/Component/Validator/Constraints/AbstractComparison.php]: Well, tests are green, no ? 😄 

More seriously, it appears that the constructors of the comparison constraints are not tested atm.

**Review by ogizanagi — APPROVED** [review]

**Comment by ogizanagi** [comment]
Hmm, it should be fixed in 3.4 actually.

**Comment by Lctrs** [comment]
`Symfony\Component\Validator\Exception\LogicException` has been introduced in `4.2`.

**Review by xabbuh — APPROVED** [review]

**Comment by fabpot** [comment]
Good catch, thanks @Lctrs.

**Pull request merged**